src/plugProjectNishimuraU/HibaState.cpp

```diff
@@ -1,529 +1,159 @@
+void StateWait::cleanup(EnemyBase*) { }
 
 /*
  * --INFO--
  * Address:	8026B6E0
  * Size:	000048
  */
-void Hiba::StateAttack::init(Game::EnemyBase*, Game::StateArg*)
+void StateAttack::init(EnemyBase* enemy, StateArg* stateArg)
 {
-	/*
-	stwu     r1, -0x10(r1)
-	mflr     r0
-	lfs      f0, lbl_8051AFF8@sda21(r2)
-	li       r5, 0
-	stw      r0, 0x14(r1)
-	stw      r31, 0xc(r1)
-	mr       r31, r4
-	li       r4, 1
-	stfs     f0, 0x2c4(r31)
-	mr       r3, r31
-	bl       startMotion__Q24Game9EnemyBaseFiPQ28SysShape14MotionListener
-	mr       r3, r31
-	bl       startFireEffect__Q34Game4Hiba3ObjFv
-	lwz      r0, 0x14(r1)
-	lwz      r31, 0xc(r1)
-	mtlr     r0
-	addi     r1, r1, 0x10
-	blr
-	*/
+	Obj* hiba     = static_cast<Obj*>(enemy);
+	hiba->m_timer = 0.0f;
+	hiba->startMotion(1, nullptr);
+	hiba->startFireEffect();
 }
 
 /*
  * --INFO--
  * Address:	8026B728
  * Size:	0000FC
  */
-void Hiba::StateAttack::exec(Game::EnemyBase*)
+void StateAttack::exec(EnemyBase* enemy)
 {
-	/*
-	stwu     r1, -0x10(r1)
-	mflr     r0
-	lfs      f0, lbl_8051AFF8@sda21(r2)
-	stw      r0, 0x14(r1)
-	stw      r31, 0xc(r1)
-	mr       r31, r4
-	stw      r30, 8(r1)
-	mr       r30, r3
-	lfs      f1, 0x200(r4)
-	fcmpo    cr0, f1, f0
-	cror     2, 0, 2
-	beq      lbl_8026B76C
-	lwz      r3, 0xc0(r31)
-	lfs      f1, 0x2c4(r31)
-	lfs      f0, 0x844(r3)
-	fcmpo    cr0, f1, f0
-	ble      lbl_8026B774
-
-lbl_8026B76C:
-	mr       r3, r31
-	bl       finishMotion__Q24Game9EnemyBaseFv
-
-lbl_8026B774:
-	lwz      r4, sys@sda21(r13)
-	mr       r3, r31
-	lfs      f1, 0x2c4(r31)
-	lfs      f0, 0x54(r4)
-	fadds    f0, f1, f0
-	stfs     f0, 0x2c4(r31)
-	bl       updateEfxLod__Q34Game4Hiba3ObjFv
-	mr       r3, r31
-	bl       interactFireAttack__Q34Game4Hiba3ObjFv
-	lwz      r3, 0x188(r31)
-	lbz      r0, 0x24(r3)
-	cmplwi   r0, 0
-	beq      lbl_8026B80C
-	lwz      r0, 0x1c(r3)
-	cmplwi   r0, 0x3e8
-	bne      lbl_8026B80C
-	lfs      f1, 0x200(r31)
-	lfs      f0, lbl_8051AFF8@sda21(r2)
-	fcmpo    cr0, f1, f0
-	cror     2, 0, 2
-	bne      lbl_8026B7EC
-	mr       r3, r30
-	mr       r4, r31
-	lwz      r12, 0(r30)
-	li       r5, 0
-	li       r6, 0
-	lwz      r12, 0x1c(r12)
-	mtctr    r12
-	bctrl
-	b        lbl_8026B80C
-
-lbl_8026B7EC:
-	mr       r3, r30
-	mr       r4, r31
-	lwz      r12, 0(r30)
-	li       r5, 1
-	li       r6, 0
-	lwz      r12, 0x1c(r12)
-	mtctr    r12
-	bctrl
-
-lbl_8026B80C:
-	lwz      r0, 0x14(r1)
-	lwz      r31, 0xc(r1)
-	lwz      r30, 8(r1)
-	mtlr     r0
-	addi     r1, r1, 0x10
-	blr
-	*/
+	Obj* hiba = static_cast<Obj*>(enemy);
+	if ((hiba->m_health <= 0.0f) || (hiba->m_timer > static_cast<Parms*>(hiba->m_parms)->m_properParms.m_activeTime.m_value)) {
+		hiba->finishMotion();
+	}
+
+	hiba->m_timer += sys->m_secondsPerFrame;
+	hiba->updateEfxLod();
+	hiba->interactFireAttack();
+
+	if ((hiba->m_animKeyEvent->m_running)
+	    && ((u32)hiba->m_animKeyEvent->m_type == 0x3E8) /* Epoch: wtf is this, needs cleanup. Surely an enum? */) {
```

Comment:
yeah this is worth looking into eventually - this comes up a lot with state stuff/in enemyBase.cpp, but I'm not sure we've pinned down what event types are what yet. should keep an eye out going forward